IPFS banner/thumbnail upload functionality

wackerow opened the pull request:
- Uses `ipfs-mini`
- `ipfs.addJSON()` used to upload and generate hash
- `ipfs.catJSON()` to fetch image from hash
- Hash and image data stored within the form state data (within `form.image.banner` and `form.image.thumbnail`)
- Forces user to upload, vs loading from an existing hash

Still needs work on styling / layout
@ryancreatescopy @samajammin 
Some questions... there are "recommended" sizes, but are we enforcing any limits to make them easier and more predictable when displaying them? And should we be limiting different file types? (ie. GIFs?)...

wackerow committed:
IPFS upload and display

wackerow committed:
removes entering IPFS hash as option

forces upload. this expands the styling and drafts displaying the images below the form

wackerow committed:
remove console log

wackerow committed:
removes requiresUpload

wackerow committed:
adds hash getters

after uploading, hashes are stored and should pass validation

ryancreatescopy commented:
I don't have a problem with users uploading gifs – perhaps there's a technical reason this shouldn't happen... But I think `.png` `.jpg` `.jpeg` and `.gif` should cover us. If so we should probably have the recognised file types in the input-description.

In regards to the dimensions, this is purely placeholder right now. Not sure whether we want to validate based on this. I did notice that a lot of the clr fund banners are pretty bad at being centred etc. Is there a way that we could calculate the optimum banner size based on aspect ratio or something?

**Design stuff**

I'm about to commit a change with a loader. I think we should do the following with loading states:
- display loader in preview box when the user hits "upload" until the image appears
  - should we even display a preview box before the user interacts with the uploader – feel like we may not need it...
- change "upload" button copy to "uploading..."

To make it clear you can overwrite an image, we should either:
- add a remove button **OR**
- change file selection button copy to "choose new file" once an image has been uploaded

re; IPFS
- Should we add a copy button to the IPFS hash so users can save their hash?
- Also should we add a tooltip to explain what IPFS is for the casual user?

finally, errors:
- what kind of errors can we come across with this process so we can write the error states?

Perhaps we pair on this?

Commit:
added loader to ipfs upload

samajammin commented:
This is looking solid - nice work!

I agree with all of @ryancreatescopy's points. 

Another general thought: given that we use this twice, we should break this out into a component to isolate this logic w/ IPFS & the uploading/loading/error/success states. The component will probably want to receive a callback after the image is uploaded so the form component can store the image hash.

Happy to pair on this early next week.

samajammin reviewed (changes requested):
See comments above.

samajammin committed:
Refactor IPFS inputs into IpfsForm component

wackerow committed:
ipfs error handling for non-images

ts linting

wackerow committed:
adds "Clear" button to start over for that image

allows user to interrupt the uploading process for whatever reason. can clear image data at any point, and can also overwrite with a new image without clearing first. Upload button disabled while loading, if there was an error on the selected image, or if there is no image selected (`document`)

wackerow committed:
copy button for IPFS hash

samajammin reviewed (approved):
Merging for now to bring in changes for other PR.

@ryancreatescopy please test this out & feel free to open a follow-up PR w/ any changes, e.g. style updates. I think the 1 item we didn't address is a tooltip / explanation on IPFS hashes for the casual user: https://www.notion.so/efdn/Figure-out-IPFS-image-uploads-8bbc1e291c2d494f87656313c43910f9
